src/htm/algorithms/SpatialPooler.hpp

```diff
@@ -148,20 +156,20 @@ class SpatialPooler : public Serializable
         columns the same regardless of the size of their receptive
         fields.
 	If numActivePerInhArea is specified then 
-	localAreaDensity must be < 0, and vice versa.
+	@ref localAreaDensity must be < 0, and vice versa.
 
-  @param stimulusThreshold This is a number specifying the minimum
+  @param stimulusThreshold This is a number specifying the minimum //TODO replace with `robustness` 0..1.0, which will affect this & synPermInc/Dec
         number of synapses that must be active in order for a column to
         turn ON. The purpose of this is to prevent noisy input from
         activating columns.
 
-  @param synPermInactiveDec The amount by which the permanence of an
+  @param synPermInactiveDec The amount by which the permanence of an //TODO make fixed and only depend on robustness?
```

Comment:
make fixed? and depend only on robustness modifier (robustness + -> both changes - )

Reply:
`synPermActiveInc` and `synPermInactiveDec` can be reformulated as `learningRate` and `coincidenceThreshold` where:
- `coincidenceThreshold = inc / dec`
- `learningRate = 1 / inc`  which is the maximum number of cycles it takes for a synapses permanence to saturate.

Reply:
that might be better, for sure learningRate. 
what is the meaning of coinc threshold? I take it it's whether it's easier to learn new (and "fill" mem faster), or forget when not repeated (so stable vs "one-shot" patterns?), vs balanced when equal (is this a golden middle?)

So for long timeseries, I'd choose to have more of forgetting, and for short, new, relatively rare events more learning? 
The SP does not really unlearn (it could, but the capacity is just huge)?